tests/queries/0_stateless/01943_non_deterministic_order_key.sql

```diff
@@ -1,3 +1,5 @@
+-- Tags: backward-incompatible
```

Comment:
Would be great to have a version guard for these tags. Eg we mark it today as backward-incompatible vs latest release. However, if we break backwards compatibility for this test in the feature it should fail.